api/storage_service.cc

```diff
@@ -1176,11 +1176,14 @@ static
 future<json::json_return_type>
 rest_set_trace_probability(std::unique_ptr<http::request> req) {
         auto probability = req->get_query_param("probability");
-        apilog.info("set_trace_probability: probability={}", probability);
-        return futurize_invoke([probability] {
+        auto write_on_close = req->get_query_param("write_on_close");
+        apilog.info("set_trace_probability: probability={} write_on_close={}", probability, write_on_close);
+        bool write_on_close_value = strcasecmp(write_on_close.c_str(), "true") == 0;
```

Comment:
I must notice, that in storage_service.cc there are 5 different ways to convert string param into boolean, and thank you for not introducing the 6th one.

Please, use `validate_bool()`